.github/workflows/main.yml

```diff
@@ -0,0 +1,61 @@
+name: Test xen-api
+
+on: [push, pull_request]
+
+jobs:
+  python-test:
+    name: Python tests
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        test: ["", "-3"]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Run python tests
+        run: bash ./.travis-python-nosetests${{ matrix.test }}.sh
+
+  ocaml-test:
+    name: Ocaml tests
+    runs-on: ubuntu-20.04
+    env:
+      package: "xapi"
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Pull configuration from xs-opam
+        run: |
+          curl https://raw.githubusercontent.com/psafont/xs-opam/gh-act/tools/xs-opam-ci.env 2> /dev/null | cut -f2 -d " " > .env
+
+      - name: Load environment file
+        id: dotenv
+        uses: falti/dotenv-action@v0.2.4
+
+      - name: Use ocaml
+        uses: avsm/setup-ocaml@v1
```

Comment:
I think this includes quite a large opam repo that we later remove - so the download it wasteful in that regard.

Reply:
This would need a modification of the action to support replacing the repository before creating a switch.

Reply:
I realise that this is an external action that relies on outside containers - just wanted to point out the inefficiency here that is undesirable.

Reply:
I understand, I also though the same when I was writing the file. I was pointing to the solution to solve the issue you brought up.

Reply:
I think Anil would be happy if he receives a PR for the customisation (or an issue to start the discussion)

Reply:
@mseri speaks accurately, as always :-)

Reply:
See https://github.com/avsm/setup-ocaml/pull/27